### ISSUE TITLE
find_package doesn't find pugixml:x64-linux

### ISSUE THREAD

**Issue by josuegomes** [microsoft/vcpkg#15915]
CMake find_package fails while trying to find pugixml

OS: Ubuntu 20.04.1
CMake 3.16

To reproduce:

1. git clone https://github.com/microsoft/vcpkg.git vcpkg
2. cd vcpkg
3. ./bootstrap.sh
4. ./vcpkg install pugixml
5. cd 
6. mkdir pugixml (files attached)
7. mkdir build
8. cd build
9. cmake .. -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE=<path to vcpkg/scripts/buildsystems/vcpkg.cmake>

Error:

```
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /home/josue/Projects/vcpkg/scripts/buildsystems/vcpkg.cmake:501 (_add_library):
  _add_library cannot create ALIAS target "pugixml" because target
  "pugixml::pugixml" is imported but not globally visible.
Call Stack (most recent call first):
  /home/josue/Projects/vcpkg/installed/x64-linux/share/pugixml/pugixml-config.cmake:16 (add_library)
  /home/josue/Projects/vcpkg/scripts/buildsystems/vcpkg.cmake:646 (_find_package)
  CMakeLists.txt:6 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/josue/Projects/pugixml/build/CMakeFiles/CMakeOutput.log".
```

Project files attached. 

[pugixml.zip](https://github.com/microsoft/vcpkg/files/5882432/pugixml.zip)

Edit: format

**Comment by josuegomes** [comment]
Output of `vcpkg install pugixml`

```
$ ./vcpkg install pugixml
Computing installation plan...
The following packages will be built and installed:
    pugixml[core]:x64-linux -> 1.11.1
Detecting compiler hash for triplet x64-linux...
Could not locate cached archive: /home/josue/.cache/vcpkg/archives/fb/fbc8177d39606f97f7990aedc70970267bbe5dea.zip
Starting package 1/1: pugixml:x64-linux
Building package pugixml[core]:x64-linux...
-- Downloading https://github.com/zeux/pugixml/archive/v1.11.1.tar.gz -> zeux-pugixml-v1.11.1.tar.gz...
-- Cleaning sources at /home/josue/Projects/vcpkg/buildtrees/pugixml/src/v1.11.1-a1ba14ec87.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/josue/Projects/vcpkg/downloads/zeux-pugixml-v1.11.1.tar.gz
-- Using source at /home/josue/Projects/vcpkg/buildtrees/pugixml/src/v1.11.1-a1ba14ec87.clean
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Fixing pkgconfig file: /home/josue/Projects/vcpkg/packages/pugixml_x64-linux/lib/pkgconfig/pugixml.pc
-- Fixing pkgconfig file: /home/josue/Projects/vcpkg/packages/pugixml_x64-linux/debug/lib/pkgconfig/pugixml.pc
-- Installing: /home/josue/Projects/vcpkg/packages/pugixml_x64-linux/share/pugixml/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: /home/josue/.cache/vcpkg/archives/fb/fbc8177d39606f97f7990aedc70970267bbe5dea.zip
Building package pugixml[core]:x64-linux... done
Installing package pugixml[core]:x64-linux...
Installing package pugixml[core]:x64-linux... done
Elapsed time for package pugixml:x64-linux: 9.224 s

Total elapsed time: 9.479 s

The package pugixml:x64-linux provides CMake targets:

    find_package(pugixml CONFIG REQUIRED)
    target_link_libraries(main PRIVATE pugixml pugixml::static pugixml::pugixml)
```

**Comment by NancyLi1013** [comment]
Hi @josuegomes 

Thanks for posting this issue.

Confirmed. This is a port bug, which has been fixed in the latest version. I will update it to the new release to solve this problem.


Related issue from upstream https://github.com/zeux/pugixml/issues/393

**Comment by NancyLi1013** [comment]
@josuegomes 

Could you please help test if the new version can work fine for you?

Thanks.

**Comment by josuegomes** [comment]
@NancyLi1013 version 1.11.4 works fine me. Note however I didn't try your PR. I have just updated the port on my own.
Notice that the hash code is different from your PR. Is that expected?

```
diff --git a/ports/pugixml/CONTROL b/ports/pugixml/CONTROL
index 65e673249..910ad8329 100644
--- a/ports/pugixml/CONTROL
+++ b/ports/pugixml/CONTROL
@@ -1,4 +1,4 @@
 Source: pugixml
-Version: 1.11.1
+Version: 1.11.4
 Homepage: https://github.com/zeux/pugixml
 Description: Light-weight, simple and fast XML parser for C++ with XPath support
diff --git a/ports/pugixml/portfile.cmake b/ports/pugixml/portfile.cmake
index 72381eb5d..bef3ce1ab 100644
--- a/ports/pugixml/portfile.cmake
+++ b/ports/pugixml/portfile.cmake
@@ -1,15 +1,16 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO zeux/pugixml
-    REF v1.11.1
-    SHA512 94a79a28d96e763cdd8951c4d0b20aefb43cd1b32cbf5a5354b09f2636710e960e2dbfa56534b61c8d6ddecb2126a006e5f485c1465a97376e1cc077df25f16e
+    REF v1.11.4
+    SHA512 a1fdf4cbd744318fd339362465472279767777b18a3c8c7e8618d5e637213c632bf9dd8144d16ae22a75cfbde007f383e2feb49084e681c930fc89a2e3f2bc4f
     HEAD_REF master
 )
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
-    OPTIONS -DUSE_POSTFIX=ON
+    OPTIONS
+        -DUSE_POSTFIX=ON
 )
 
 vcpkg_install_cmake()

```

**Comment by NancyLi1013** [comment]
Thanks for your test and feedback. It's so great.

I think it should be related with the `REF `. I used the commit id, you used the tag name. So the hash code is different.

Both of them should be okay. Let's wait for the PR #https://github.com/microsoft/vcpkg/pull/15207 merged.

**Comment by NancyLi1013** [comment]
Closing this issue since it has been solved via PR #15207.

**Comment by josuegomes** [comment]
Confirmed solved after PR
